packit_service/worker/parser.py

```diff
@@ -601,6 +617,111 @@ def parse_pull_request_comment_event(
             comment=comment,
         )
 
+    @staticmethod
+    def parse_check_rerun_event(
+        event,
+    ) -> Optional[
+        Union[CheckRerunCommitEvent, CheckRerunPullRequestEvent, CheckRerunReleaseEvent]
+    ]:
+        """Look into the provided event and see if it is Github check rerun event."""
+        if not (
+            nested_get(event, "check_run")
+            and nested_get(event, "action") == "rerequested"
+        ):
+            return None
+
+        check_name = nested_get(event, "check_run", "name")
+        logger.info(f"Github check run {check_name} rerun event.")
+
+        deployment = ServiceConfig.get_service_config().deployment
+        app = nested_get(event, "check_run", "app", "slug")
+        if (deployment == Deployment.prod and app != "packit-as-a-service") or (
+            deployment == Deployment.stg and app != "packit-as-a-service-stg"
+        ):
+            logger.warning("Check run not created by this app.")
+            return None
+
+        check_name_job, check_name_target = None, None
+        check_name_split = check_name.split("/", maxsplit=2)
+        if len(check_name_split) == 2:
+            check_name_job_info = check_name_split[1]
+            for job_name in MAP_CHECK_PREFIX_TO_HANDLER.keys():
+                if check_name_job_info.startswith(job_name):
+                    check_name_job = job_name
+                    # e.g. [testing-farm-]fedora-rawhide-x86_64
+                    check_name_target = check_name_job_info[
+                        (len(job_name) + 1) :  # noqa
+                    ]
+
+        if not (check_name_job and check_name_target):
+            logger.warning(
+                "We were not able to parse the job and target from the check run name."
+            )
+            return None
```

Comment:
also log the variables please (even though they will be visible in sentry, but still, let's have them in logs as well)